README.md

```diff
@@ -1,104 +1,166 @@
+   zenoh::net::runtime: Using ZID: f7bc54e0941036422ec08ebac6fbdb40
+   zenoh::api::loader: Loading  plugin "remote_api"
+   zenoh::api::loader: Starting  plugin "remote_api"
+   zenoh::api::loader: Successfully started plugin remote_api from "/usr/lib/libzenoh_plugin_remote_api.so"
+   zenoh::api::loader: Finished loading plugins
+   zenoh::net::runtime::orchestrator: Zenoh can be reached at: tcp/....
+   ```
 
-3. Run `cargo build`
+1. Build both the plugin and the router from the sources:
 
-## **Examples of usage**
+   Build the plugin `zenoh-plugin-remote-api`
 
-### Running the Rust Plugin
+   ```sh
+   cargo build 
+   ```
 
-Prerequisites:
- - You have a zenoh router (`zenohd`) installed, and the `zenoh_plugin_remote_api` library file is available in `~/.zenoh/lib`.
+   Build and run the zenohd from the same sources which were used for the plugin.
+   The zenohd dependency is specified in the `[workspace.metadata.bin]` section in Cargo.toml which is processed by the 3rd party tool [cargo-run-bin](https://crates.io/crates/cargo-run-bin).
 
-### **Setup via a JSON5 configuration file**
+   ```sh
+   cargo install cargo-run-bin
+   cargo bin zenohd --config EXAMPLE_CONFIG.json5
+   ```  
 
-  - Create a `zenoh.json5` configuration file containing for example:
-    ```json5
-    {
-      plugins: {
-        // configuration of "storage_manager" plugin:
-        remote_api: {
-          "websocket_port": "10000",
-          // secure_websocket configuration is optional
-          "secure_websocket": {
-                "certificate_path" : "/path/to/certificate",
-                "private_key_path" : "/path/to/private_key"
-          }
-        }
-        // Optionally, add the REST plugin
-        rest: { http_port: 8000 }
-      }
-    }
-    ```
-  - Run the zenoh router with:
-    `zenohd -c zenoh.json5`
+   Expected output is like:
 
-## Adding Typescript to your application
+   ```txt
+   zenohd: zenohd vc764bf9b built with rustc 1.75.0 (82e1608df 2023-12-21)
+   zenoh::net::runtime: Using ZID: bb3fb16628f57e92f92accf2f5c81511
+   zenoh::api::loader: Loading  plugin "remote_api"
+   zenoh::api::loader: Starting  plugin "remote_api"
+   zenoh::api::loader: Successfully started plugin remote_api from "./target/debug\\zenoh_plugin_remote_api.dll"
+   zenoh::api::loader: Finished loading plugins
+   zenoh::net::runtime::orchestrator: Zenoh can be reached at: tcp/...
+   ```
 
-The TypeScript library can be install from the command line: 
+### Building the library
 
-`npm install @eclipse-zenoh/zenoh-ts@0.0.8`
+1. Make sure that the following utilities are available on your platform.
 
-Or added via package.json
+- [NPM](https://www.npmjs.com/package/npm)
+- [yarn](https://classic.yarnpkg.com/lang/en/docs/install/#debian-stable)
+- [Typescript](https://www.typescriptlang.org/download/) 
 
-`"@eclipse-zenoh/zenoh-ts": "0.0.8" `
+2. Navigate to the directory `zenoh-ts`
+
+1. Run the commands:
```

Comment:
```suggestion
1. Navigate to the directory `zenoh-ts`

2. Run the commands:
```

Reply:
fixed - it all shoud be 1. 1. 1. - md does numbering automatically